cpp/src/strings/convert/convert_urls.cu

```diff
@@ -0,0 +1,279 @@
+
+    auto strings_column = column_device_view::create(strings.parent(), stream);
+    auto d_strings = *strings_column;
+
+    // copy null mask
+    rmm::device_buffer null_mask = copy_bitmask(strings.parent(),stream,mr);
+    // build offsets column
+    auto offsets_transformer_itr = thrust::make_transform_iterator( thrust::make_counting_iterator<size_type>(0),
+                                                                    url_encoder_fn{d_strings} );
+    auto offsets_column = make_offsets_child_column(offsets_transformer_itr,
+                                                    offsets_transformer_itr+strings_count,
+                                                    mr, stream);
+    auto d_offsets = offsets_column->view().data<int32_t>();
+    // build chars column
+    auto chars_column = create_chars_child_column( strings_count, strings.null_count(),
+                                                   thrust::device_pointer_cast(d_offsets)[strings_count],
+                                                   mr, stream );
+    auto d_chars = chars_column->mutable_view().data<char>();
+    thrust::for_each_n(rmm::exec_policy(stream)->on(stream),
+                       thrust::make_counting_iterator<size_type>(0), strings_count,
+                       url_encoder_fn{d_strings,d_offsets,d_chars});
+    //
+    return make_strings_column(strings_count, std::move(offsets_column), std::move(chars_column),
+                               strings.null_count(), std::move(null_mask), stream, mr);
+}
+
+} // namespace detail
+
+// external API
+std::unique_ptr<column> url_encode( strings_column_view const& strings,
+                                    rmm::mr::device_memory_resource* mr)
+{
+    return detail::url_encode(strings,mr);
+}
+
+namespace detail
+{
+namespace
+{
+
+//
+// This is the functor for the url_decode() method below.
+// Specific requirements are documented in custrings issue #321.
+// In summary it converts all character sequences starting with '%' into bytes
+// interpretting the following 2 characters as hex values to create the output byte.
+// For example, the sequence '%20' is converted into byte (0x20) which is a single
+// space character. Another example converts '%C3%A9' into 2 sequential bytes
+// (0xc3 and 0xa9 respectively). Overall, 3 characters are converted into one byte
+// whenever a '%' character is encountered in the string.
+//
+struct url_decoder_fn
+{
+    column_device_view const d_strings;
+    int32_t const* d_offsets{};
+    char* d_chars{};
+
+    // utility to convert 2 hex chars into a single byte
+    __device__ char hex_to_byte( char ch1, char ch2 )
+    {
+        u_char result = 0;
+        if( ch1 >= '0' && ch1 <= '9' )
+            result += (ch1-48);
+        else if( ch1 >= 'A' && ch1 <= 'Z' )
+            result += (ch1-55);
+        else if( ch1 >='a' && ch1 <= 'z' )
+            result += (ch1-87);
+        result *= 16;
+        if( ch2 >= '0' && ch2 <= '9' )
```

Comment:
```suggestion
        if(std::isdigit(ch2))
```

Reply:
Cannot call `std::isdigit` this in device code.